README.md

```diff
@@ -28,21 +32,37 @@ Also check the [config.yaml documentation](https://github.com/haskell/ghcup-hs/b
 
 ## For contributors
```

Comment:
After adding a section for "upstream devs" (below), this title sounds a little misleading, because upstream devs are also contributors in some sense. I'd do away with these vague terms and put in titles something less ambiguous. E.g. "Contributing to GHCup main channel" and "Contributing to GHCup vanilla channel (for upstream developers)" or something. Although, this current section is more about how the main channel is set up rather than how to contribute to it. And that's a good thing: we need first to explain what we have here before explaining how to contribute. It's just that the title doesn't quite match.

Having separate sections for how each channel operates before sections explaining step-by-step what to do to contribute would be a great improvement! I'd also expect the step-by-step part to be different for vanilla vs. the main channel because, hopefully, updating vanilla should involve less cooperation from GHCup's side. At least, no bindists. Optionally, no GPG.

Reply:
I thought that depending on a channel you want to contribute to, the requirements are different. If that's the case, this should be made more explicit. If it's not the case, then nevermind... At least, the vanilla channel is special in that it doesn't involve building bindists in those separate repositories that you mention at the top (`[stable-haskell/cabal](https://github.com/stable-haskell/cabal)` and others). This should be spelled out: this is a **major** difference!